src/transformers/generation/configuration_utils.py

```diff
@@ -908,6 +908,13 @@ def to_json_string(self, use_diff: bool = True, ignore_metadata: bool = False) -
             for metadata_field in METADATA_FIELDS:
                 config_dict.pop(metadata_field, None)
 
+        def convert_keys_to_string(dictionary):
+            if not isinstance(dictionary, dict):
+                return dictionary
+            return {str(key): convert_keys_to_string(value) for key, value in dictionary.items()}
```

Comment:
Let's use a clearer name than `dictionary` here - since it might not be a dictionary

```suggestion
        def convert_keys_to_string(obj):
            if not isinstance(obj, dict):
                return obj
            return {str(key): convert_keys_to_string(value) for key, value in obj.items()}
```